src/main/java/net/snowflake/ingest/streaming/internal/ParquetTypeInfo.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.Map;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+
+/**
+ * Util class that contains Parquet type and other metadata for that type needed by the Snowflake
+ * server side scanner
+ */
+class ParquetTypeInfo {
+  private Type parquetType;
+  private Map<String, String> metadata;
+
+  public Type getParquetType() {
+    return this.parquetType;
+  }
+
+  public Map<String, String> getMetadata() {
+    return this.metadata;
+  }
+
```

Comment:
lets introduce a constructor to this class that forces the callsite to provide both parquetType and metadata at construction time. I checked the one callsite that constructs PTI and it is always calling both setters (at the end of ParquetTypeGenerator.generateColumnParquetTypeInfo), so might as well enforce at construction time than have this randomly mutable state on this object.

Also mark the two private fields in this class as final once you do this.

Reply:
Created ctor and removed setters.